src/io.c

```diff
@@ -162,16 +162,20 @@ void io_init(const char *cfile, const char *datadir, const char *confdir)
 		asprintf(&path_ddir, "%s%s", datadir, "/");
 
 	if (!confdir)
-		if (home_dir) {
-			asprintf(&path_cdir, "%s%s", home_dir, "/" DIR_NAME_LEGACY);
-			if (!io_dir_exists(path_cdir)) {
-				if ((path_cdir = getenv("XDG_CONFIG_HOME")))
-					asprintf(&path_cdir, "%s%s", path_cdir, "/" DIR_NAME);
-				else
-					asprintf(&path_cdir, "%s%s", home_dir, "/.config/" DIR_NAME);
+		if(!datadir) {
+			if (home_dir) {
+				asprintf(&path_cdir, "%s%s", home_dir, "/" DIR_NAME_LEGACY);
+				if (!io_dir_exists(path_cdir)) {
+					if ((path_cdir = getenv("XDG_CONFIG_HOME")))
+						asprintf(&path_cdir, "%s%s", path_cdir, "/" DIR_NAME);
+					else
+						asprintf(&path_cdir, "%s%s", home_dir, "/.config/" DIR_NAME);
+				}
+			} else {
+				path_cdir = "./." DIR_NAME;
```

Comment:
Shouldn't this whole change be squashed into the first patch that touches io_init()? Also need to address memory leaks, as pointed out in my review of the other patch.